proto/substrait/algebra.proto

```diff
@@ -796,7 +796,8 @@ message Expression {
       string string = 12;
       bytes binary = 13;
       // Timestamp in units of microseconds since the UNIX epoch.
-      int64 timestamp = 14;
+      // Deprecated in favor of `Timestamp timestamp`
+      // int64 timestamp = 14 [deprecated = true];
```

Comment:
Technically the number would become reserved to prevent reuse.  But usually one goes through a deprecation period to allow for interoperability.  That means we need to have both around for a while.  We could say that consumers always should use matched sets but I don't think that's always possible.

Reply:
> That means we need to have both around for a while. We could say that consumers always should use matched sets but I don't think that's always possible.

For migration purposes we need a period of time where both of these timestamps types coexist. To migrate this properly we need to be able to do something like:
1. Update consumers to handle the new timestamp type.
2. Update producers to generate plans with the new timestamp type and remove the old type.
3. Remove code for handling the old timestamp type in consumers.

If we remove this entirely we have to do 1 and 2 simultaneously, which doesn't work super well if they're entirely different systems 😢

Reply:
Makes sense.  So we'll need a different name for the new timestamp.  Maybe something like `ParametrizedTimestamp`?

Reply:
Since parameterized_types.proto already uses `Parameterized` as a prefix in all the names, it'll be less confusing to use something like `PrecisionTimestamp` instead